apps/xmtp.chat/src/components/Messages/Message.tsx

```diff
@@ -1,44 +1,35 @@
 import { Box, Flex, Paper, Stack, Text } from "@mantine/core";
-import type { Client, DecodedMessage } from "@xmtp/browser-sdk";
+import type { Client, Conversation, DecodedMessage } from "@xmtp/browser-sdk";
+import {
+  ContentTypeTransactionReference,
+  type TransactionReference,
+} from "@xmtp/content-type-transaction-reference";
+import {
+  ContentTypeWalletSendCalls,
+  type WalletSendCallsParams,
+} from "@xmtp/content-type-wallet-send-calls";
 import { intlFormat } from "date-fns";
-import { useNavigate, useOutletContext } from "react-router";
+import { useOutletContext } from "react-router";
 import { shortAddress } from "@/helpers/address";
 import { nsToDate } from "@/helpers/date";
-import classes from "./Message.module.css";
 import { MessageContent } from "./MessageContent";
+import { TransactionReferenceUI } from "./TransactionReference";
+import { WalletSendCallsUI } from "./WalletSendCalls";
 
 export type MessageProps = {
   message: DecodedMessage;
+  sendMessage: Conversation["send"];
 };
 
-export const Message: React.FC<MessageProps> = ({ message }) => {
+export const Message: React.FC<MessageProps> = ({ message, sendMessage }) => {
   const { client } = useOutletContext<{ client: Client }>();
```

Comment:
it's no longer necessary to pass down the `sendMessage` method. you should be able to access the conversation directly from the outlet context like so:

```
const { client, conversation } = useOutletContext<ConversationOutletContext>();
```